vmm/sandbox/src/stratovirt/mod.rs

```diff
@@ -446,15 +454,18 @@ impl StratoVirtVM {
         ))
     }
 
-    fn attach_to_pcie_rootbus<T: StratoVirtDevice + Sync + Send + 'static>(
+    fn attach_to_bus<T: StratoVirtDevice + Sync + Send + 'static>(
         &mut self,
         mut device: T,
     ) -> Result<()> {
-        // get the empty slot index
-        let slot_index = self.get_empty_pcie_slot_index()?;
-        // set the pcie slot status to Occupied
-        self.pcie_root_bus.bus.slots[slot_index].status = SlotStatus::Occupied(device.id());
-        device.set_device_addr(slot_index);
+        if self.pcie_root_bus.is_some() {
+            // get the empty slot index
+            let slot_index = self.get_empty_pcie_slot_index()?;
+            // set the pcie slot status to Occupied
+            self.pcie_root_bus.as_mut().unwrap().bus.slots[slot_index].status =
+                SlotStatus::Occupied(device.id());
+            device.set_device_addr(slot_index);
+        }
```

Comment:
```suggestion
        if let Some(&mut pcie_root_bus) = self.pcie_root_bus {
            ...
            pcie_root_bus.bus.slots[slot_index].status =
                SlotStatus::Occupied(device.id());
            ...
        }
```
I am not sure does it work? :)

Reply:
I tried but it did not work well... 
move occurs because `pcie_root_bus`  does not implement the `Copy` trait, but `self.get_empty_pcie_slot_index` needs to use it next